### PR TITLE
gitea: fetch all branches

### PR DESCRIPTION
**Pull request by ennetech**
Both repo branch view and manual branch selector were limited to the stock pageSize of gitea client.

I've added a loop to fetch all branches from gitea

### CODE REVIEW

File: server/remote/gitea/gitea.go [hunk 1]
```diff
@@ -462,15 +462,28 @@ func (c *Gitea) Branches(ctx context.Context, u *model.User, r *model.Repo) ([]s
...
+		if err != nil {
+			return nil, err
+		}
+		if len(giteaBranches) > 0 {
+			for _, branch := range giteaBranches {
+				branches = append(branches, branch.Name)
+			}
+			page++
+		} else {
+			page = -1
```
Comment (6543) [inline server/remote/gitea/gitea.go:483]: use break ?

Comment (ennetech) [reply server/remote/gitea/gitea.go]: solved

**Comment by ennetech** [comment]
@qwerty287 i'll add the todo, do you prefer it in the base interface or just in the function modified?

**Comment by 6543** [comment]
well just make a func that get a func as argument in common ... witht is you dont need to create special interfaces etc ...

just some func as arg and generics:
```go
func pagginate[T type](get func(page) ([]*T, error)) ([]*T, error)
```

?

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@qwerty287 did you have something like this in mind: #1236 ?

**Comment by 6543** [comment]
@ennetech :point_up: that's a nice example how to use generics and pass func as arguments in go

**Comment by 6543** [comment]
Yes if we start supporting pagination ... we do not need that func in most cases anymore

**Pull request merged**